webrender/src/renderer.rs

```diff
@@ -740,10 +742,29 @@ impl Renderer {
         // Pull any pending results and return the most recent.
         while let Ok(msg) = self.result_rx.try_recv() {
             match msg {
-                ResultMsg::UpdateTextureCache(update_list) => {
+                ResultMsg::UpdateTextureData(update_list, external_image_update_list) => {
                     self.pending_texture_updates.push(update_list);
+
+                    if self.external_image_handler.is_some() {
+                        self.pending_external_image_updates.push(external_image_update_list);
+                    }
                 }
                 ResultMsg::NewFrame(frame, profile_counters) => {
+                    // When a new frame is ready, we could start to update all pending external
+                    // image request here.
+                    if !self.pending_external_image_updates.is_empty() {
```

Comment:
api.add_image(A)
dl.push_image() with A
set_root_dl(frame 1)
update()
render() <= the frame 1 is not ready, draw frame 0 instead
api.delete_image(A)
api.add_image(B)
dl.push_image with B
update()
render() <= the frame 1 is ready now, draw with A
-----
So, we the image A will still be used until frame 2 ready. That's why I put the release() call when we receives the new frame.